pkg/account/domain/account.go

```diff
@@ -151,13 +155,55 @@ func (a *AccountV2) DeleteSearchFilter(id string) error {
 	return errSearchFilterNotFound
 }
 
-func (a *AccountV2) UpdateSearchFilter(searchFilter *proto.SearchFilter) error {
-	for i, f := range a.AccountV2.SearchFilters {
-		if f.Id == searchFilter.Id {
-			a.AccountV2.SearchFilters[i] = searchFilter
+func (a *AccountV2) UpdateDefaultSearchFilter(id string, defaultFilter bool) error {
+	for _, f := range a.AccountV2.SearchFilters {
+		if f.Id == id {
+			// Since there is only one default setting for a filter target, set the existing default to OFF.
+			if defaultFilter {
+				a.ternOffDefaultFilters(f.FilterTargetType, f.EnvironmentId)
+			}
+
+			f.DefaultFilter = defaultFilter
+			a.UpdatedAt = time.Now().Unix()
+			return nil
+		}
+	}
+	return errSearchFilterNotFound
+}
+
+func (a *AccountV2) UpdateSearchFilter(
```

Comment:
Do we need this function?
We have a function to change the default filter.
If you plan to update the name and query for the existing filters, we should use `changeName` and `changeQuery` instead.